include/n64dd.h

```diff
@@ -4,17 +4,20 @@
 #include "ultra64.h"
 #include "z64pause.h"
 #include "z64scene.h"
+#include "z64map_mark.h"
 
 struct Font;
 struct GameState;
-union Gfx;
+struct MapData;
 struct MessageTableEntry;
 struct PlayState;
 struct RegEditor;
 struct RoomContext;
 struct SaveContext;
 struct Scene;
 
+union Gfx;
```

Comment:
This file includes the libultra headers already, there should be no need to forward declare `Gfx`, or am I missing something?

Reply:
I probably just pattern-matched on the rest, I removed it. Tbh I don't like forward declaring these structs instead of importing the typedefs, it feels inconsistent. It doesn't work for everything (e.g. typedefs that are not structs) and it's slightly jarring to read in function signatures

Reply:
I understand this reasoning, but forward declaring allows us to improve the awful situation of almost every header including each other, which would lead to circular dependencies.


I guess a possible alternative would be to just have other headers do the typedefs and remove the typedefs from the actual structs.
```c
// typdefs.h or something
typedef struct PlayState PlayState;
typedef struct RoomContext RoomContext;
```
But I think it would be convoluted and annoying to use with IDEs, like doing a "Go to definition" on a type would lead you to the typedef instead of the actual struct. I don't think it would be worth

Reply:
In this case there would be no circular dependencies, no other header should be importing n64dd stuff. Perhaps we should forward declare structs only when necessary instead of reaching for it first

Reply:
the other thing is bss stability and dependency file handling (compiler-generated .d makefiles), not just circular deps